src/components/timers/XY.tsx

```diff
@@ -1,3 +1,242 @@
+    const [repRemaining, setRepRemaining] = useState(0)
+    const intervalRef = useRef<number | null>();
+
+
+    const secondsOnTimer = secondsRemaining % 60
+    const minutesRemaining = (secondsRemaining - secondsOnTimer) / 60
+    const minutesOnTimer = minutesRemaining % 60
+    const hoursOnTimer = (minutesRemaining - minutesOnTimer) / 60
+
+    const totalSeconds = convertToSeconds(timeMinInput, timeSecInput);
+  
+    const startStopCountdown = () => {
+
+
+                if (Number.isNaN(totalSeconds) || (timeMinInput === '' && timeSecInput === '') || totalSeconds <= 0 || Number.isNaN(repInput) || (repInput === '')) {
+                    alert('Please enter a valid time.');
+                    }
+                  else if (totalSeconds > 3600) {
+                    alert("Friendly caution: excercise over an hour can lead to overtraining. Please enter a time under an hour.")
+                  }
+              else {
+                if (status !== STATUS.STARTED) {
+                    if (status===STATUS.INITIAL) {
+                        setSecondsRemaining(totalSeconds);
+                        setRepRemaining(Number(repInput));
+                    }
+                  setStatus(STATUS.STARTED);
+                } else {
+                  setStatus(STATUS.STOPPED); 
+                }
+             }
+        }
+        ;
+
+    const resetCountdown = () => {
+      setStatus(STATUS.STOPPED);
+      const totalSeconds = convertToSeconds(timeMinInput, timeSecInput);
+      setSecondsRemaining(totalSeconds);
+      setRepRemaining(Number(repInput));
+    }
+
+    const fastforwardCountdown = () => {
+        setStatus(STATUS.FASTFORWARDED);
+        setSecondsRemaining(0);
+        setRepRemaining(0);
+        if (intervalRef.current !== null) {
+            clearInterval(intervalRef.current);
+            intervalRef.current = null;
+          }
+      }
+
+    const initialCountdown = () => {
+        setStatus(STATUS.INITIAL);
+      }  
+
+      useEffect(() => {
+        if (
+          (status === STATUS.STARTED &&
+          secondsRemaining === 0 &&
+          repRemaining > 0)
+        ) {
+            const totalSeconds = convertToSeconds(timeMinInput, timeSecInput);
+            setSecondsRemaining(totalSeconds);
+            setRepRemaining(Number(repInput));
+            const reps = repRemaining - 1;
+    
+                    if (reps === 0) {
+                        setSecondsRemaining(0);
+                    setStatus(STATUS.STOPPED);
+                    
+                    } 
+                    else {
+                    setSecondsRemaining(totalSeconds);
+                    }
+          setRepRemaining(reps);
+        }
+      }, [secondsRemaining, status, repRemaining, repInput, timeMinInput, timeSecInput]);
+
+      
+      useEffect(() => {
+        if (status === STATUS.STARTED) {
+        
+            //this decrements the seconds immideately without waiting for the first second
```

Comment:
Remove the initial decrement for the XY - it's losing our first second on the first round